thunder/tests/test_interpreter.py

```diff
@@ -3572,3 +3572,12 @@ def fn():
         jfn()
     except Exception as e:
         assert 'print("A" * "A")' in "\n".join(traceback.format_tb(e.__traceback__))
+
+
+def test_binary_subscr_on_types():
+    def fn():
+        return list[int]
```

Comment:
I think we should avoid needing to have to deal with the list class getitem (which is more elaborate) for this PR.
```suggestion
    class A:
        @classmethod
        def __class_getitem__(cls, index):
            return index

    def fn():
        return A[int]
```